### PR TITLE
Table of contents nav should show where you are

### PR DESCRIPTION
**Pull request by olyism**
- [x] Added scripts to highlight current heading
- [x] Improved styling:
  - [x] Use purple theme consistent with our navigation look and feel
  - [x] Tighten spacing
- [x] Introduce accordion for small screens

Fixes to do:

- [x] Look for ways to re-use color variables
- [x] RSpec fix
- [x] `aria-current`
  
Screen Shot 2023-02-03 at 4 39 58 pm

Screen Shot 2023-02-03 at 4 40 25 pm

### CODE REVIEW

File: app/assets/javascripts/toc.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const getDistanceFromTop = (element) => {
+      if (!element.offsetParent) {
+        return element.offsetTop;
+      }
+      return element.offsetTop + getDistanceFromTop(element.offsetParent);
+    };
+
+    const getHeadingPos = (heading) => getDistanceFromTop(heading);
+
+    const setCurrentLink = (hash) => {
```
Comment (dannymidnight) [inline app/assets/javascripts/toc.js:19]: Should we consider adding an [`aria-current`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) attribute when highlighting the link?

File: app/assets/stylesheets/public/_toc.scss [hunk 2]
```diff
@@ -1,44 +1,108 @@
 .Toc {
+  background-color: #f8f8f8;
```
Comment (dannymidnight) [inline app/assets/stylesheets/public/_toc.scss:2]: Is there a variable kicking around for this one? Feels like we should be pulling a few of these settings from a single point to minimise the blast radius of potential future style changes.

**Review by dannymidnight — COMMENTED** [review]
Nice one, looks like a nice improvement.

File: app/assets/javascripts/toc.js (hunk 1 above)
Comment (olyism) [reply app/assets/javascripts/toc.js:19]: TIL `aria-current`. Great idea. I've added [this](https://github.com/buildkite/docs/pull/1851/commits/6c8639a7376e49bbe8968910889aab32b5f6cfa4#diff-2244aad9e766cfd2cf16ab727def0e5d7b36db14310f993e8f2ebf63d2a9b422R22)

File: app/assets/stylesheets/public/_toc.scss (hunk 2 above)
Comment (olyism) [reply app/assets/stylesheets/public/_toc.scss]: Agree 👍 Will start adding variables to Docs.

I've exported the color variables from our Figma Library and putted them in `_variables.scss` and assigned them in https://github.com/buildkite/docs/pull/1851/commits/e58164db0ebbea65efabe5255883bb1c0a9aed20

I've done this for `_toc.scss`, replaced all instances of the lime color, and will gradually roll this out for other hex colors in a future card to keep this PR small for now.

**Comment by dannymidnight** [comment]
Closing this branch since it landed in another changeset.

**Pull request closed without merging**